controller/src/beerocks/master/db/node.cpp

```diff
@@ -350,28 +350,19 @@ bool node::ap_metrics_data::add_ap_metric_data(std::shared_ptr<wfa_map::tlvApMet
     channel_utilization                 = ApMetricData->channel_utilization();
     number_of_stas_currently_associated = ApMetricData->number_of_stas_currently_associated();
 
-    if (ApMetricData->estimated_service_parameters().include_ac_be) {
-        std::copy(ApMetricData->estimated_service_info_field_ac_be(),
-                  ApMetricData->estimated_service_info_field_ac_be() + 3,
-                  estimated_service_info_field_ac_be);
+    auto len = ApMetricData->estimated_service_info_field_length();
+    //copy all fields to database vector
+    for (unsigned int i = 0; i < len; ++i) {
+        estimated_service_info_fields.push_back(*ApMetricData->estimated_service_info_field(i));
```

Comment:
The normal way to do this would be

```
std::copy(ApMetricData->estimated_service_info_field(),
                ApMetricData->estimated_service_info_field() + ApMetricData->estimated_service_info_field_length(),
                std::back_inserter(estimated_service_info_fields));
```